### DISCUSSION TITLE
Chunk Manipulation & Editing Tools

### DISCUSSION THREAD

**Discussion by ihategravel2** [Mojang/minecraft-editor#71] [Feedback]
*Note: This has been rewritten.*

Possible Functions:
- Delete Selected Chunks
- Copy Selected Chunks
- Regenerate Selected/All Chunks
- Start/Stop Chunk Generation
- Pruning Chunks outside of the selected area.

2D Viewer:
A 2D view of the chunks would make for a much more seamless chunk editing experience. Doing it in a Three Dimensional Space would require you flying massive distances.

 I don't think rendering the chunks in the conventional way would be practical as that would be dependent on your normal render distance and would cause performance issues and would fall short especially with a large sized world. Some sort of simplified rendering of 2d chunks seems like a possible solution... 🤷

**Comment by SmokeyStack** [comment]
Related to https://github.com/Mojang/minecraft-editor/discussions/63 ?

**Reply by ihategravel2** [reply]
I suppose it could be but I think this would be different.

**Comment by kakinnun** [comment]
Related to this, pulling comments I've seen from elsewhere:

chunk manipulation functions:

start/stop chunk generation
select ENTIRE chunks
delete ENTIRE chunks
prune chunks outside a selected area

**Comment by ihategravel2** [comment]
Chunk deletion and Chunk Regeneration were added in Preview 26.0.28

https://feedback.minecraft.net/hc/en-us/articles/42634616536589-Minecraft-Beta-Preview-26-0-28
